### PR TITLE
Make the version part of th UUID

### PR DESCRIPTION
**Pull request by Corniel**
Extended the UUID with the UUID version (4 = random, 3 = MD5 based, 5 = SHA1 based), with according factory methods.

### CODE REVIEW

File: src/Qowaiv/Uuid.cs [hunk 1]
```diff
@@ -7,436 +7,435 @@
...
+            Guard.NotNull(writer, "writer");
+            writer.WriteString(ToString(CultureInfo.InvariantCulture));
+        }
+
+        #endregion
+
+        #region (JSON) (De)serialization
+
+        /// <summary>Generates a UUID from a JSON null object representation.</summary>
+        void IJsonSerializable.FromJson()=>m_Value = default(Guid);
```
Comment (plat2059) [inline src/Qowaiv/Uuid.cs]: Nitty-gritty mode on: some whitespace around => ?

File: src/Qowaiv/Uuid.cs [hunk 2]
```diff
@@ -7,436 +7,435 @@
...
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>Generates an <see cref="Uuid"/> applying a <see cref="MD5"/> hash on the data.</summary>
+        public static Uuid GenerateWithMD5(byte[] data)
+        {
+            Guard.NotNull(data, nameof(data));
+            var hash = MD5.Create().ComputeHash(data);
```
Comment (plat2059) [inline src/Qowaiv/Uuid.cs]: Shouldn't the instance of MD5 be disposed?

File: src/Qowaiv/Uuid.cs [hunk 3]
```diff
@@ -7,436 +7,435 @@
...
+            UuidExtensions.SetVersion(hash, UuidVersion.MD5);
+            return new Guid(hash);
+        }
+
+        /// <summary>Generates an <see cref="Uuid"/> applying a <see cref="SHA1"/> hash on the data.</summary>
+        public static Uuid GenerateWithSHA1(byte[] data)
+        {
+            Guard.NotNull(data, nameof(data));
+            var bytes = SHA1.Create().ComputeHash(data);
```
Comment (plat2059) [inline src/Qowaiv/Uuid.cs]: Same here. Using?

File: src/Qowaiv/UuidExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        public static UuidVersion GetVersion(this Guid guid)
+        {
+            var bytes = guid.ToByteArray();
+            var version = bytes[7] >> 4;
+            return (UuidVersion)version;
+        }
+
+        internal static void SetVersion(byte[] uuid, UuidVersion version)
+        {
+            uuid[7] &= 0x0F;
```
Comment (plat2059) [inline src/Qowaiv/UuidExtensions.cs]: Doubting here if you should only allow changing the empty version to the passed version.

File: src/Qowaiv/Uuid.cs (hunk 2 above)
Comment (Corniel) [reply src/Qowaiv/Uuid.cs]: It should. 👍

**Pull request merged**